### PR TITLE
Update a connector

### LINKED ISSUE #145: ✨ Update a connector
> **As a** user
> **I want** want to change the configuration of a connector 
> **because** e.g the discord channel changed

> ⚠️ Blocked by #144

## Description:
With #144 a user can see a list of all connectors. Now it's time that the user can change the configuration. The following attributes can be changed:
* `active`
* `default`
* `socket` 

It should be not possible to change any other attribute! 

### 🟢 In scope:
* Modular input validation!
* More input validation!
* Implement `PATCH /api/connectors/:id`
* Show notification when a connector has been updated (with name!)

### 🔴 Not in scope:
* Implement other endpoints
* Update Documentation

### PR DESCRIPTION
**Pull request by tjarbo**
> **As a** user
> **I want** want to change the configuration of a connector 
> **because** e.g the discord channel changed

## Description
Close #145 
With this user story the user is now able to update connector settings via the frontend. The following attributes can be changed:
* `active` - Defines if a connector will be able to send messages
* `default` - Defines if a connector will be used if no connector has been found based on the related courses
* `socket` - Depends on the connector type - For Discord Bot it is `token` and `channel`

### Screenshots
grafik

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## Checklist

- [ ] ~Make sure you are requesting to main. Don't request to release branch!~
- [X] I have linked the related issue.
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Review by piuswalter — APPROVED** [review]

File: packages/backend/src/controllers/connectors/plugins/discordBot.class.ts [hunk 1]
```diff
@@ -20,9 +24,22 @@ export class DiscordBotConnectorPlugin extends ConnectorPlugin {
    */
   constructor(protected document: IConnectorDocument) {
     super();
+    this.setupClient();
+  }
+
+  private async setupClient(): Promise<void> {
```
Comment (tjarbo) [inline packages/backend/src/controllers/connectors/plugins/discordBot.class.ts:30]: Missing JSDoc

File: packages/backend/src/routes/connectors.routes.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
 import { Router } from 'express';
-import { connectorsGetRequest } from '../controllers/connectors/handlers';
+import { connectorsGetRequest, connectorsIdPatchRequest } from '../controllers/connectors/handlers';
 export const connectorsRoutes = Router();
 
 // register routes
 connectorsRoutes.get('/connectors', connectorsGetRequest);
 connectorsRoutes.post('/connectors');
 connectorsRoutes.post('/connectors/:id');
```
Comment (tjarbo) [inline packages/backend/src/routes/connectors.routes.ts:8]: Endpoint not needed

File: packages/frontend/src/components/connectors/ConnectorPanel.vue [hunk 3]
```diff
@@ -91,15 +106,48 @@ export default {
...
+        body: {
+          active: this.connectorDataChanged.active,
+          courses: this.connectorDataChanged.courses,
+          default: this.connectorDataChanged.default,
+          name: this.connectorDataChanged.name,
+          socket: this.connectorDataChanged.socket,
+        },
+      };
+
+      if (payload.body.socket.token === 'hidden') delete payload.body.socket.token;
```
Comment (tjarbo) [inline packages/frontend/src/components/connectors/ConnectorPanel.vue:130]: Make condition independent from hardcoded string

File: packages/frontend/src/components/connectors/ConnectorPanel.vue [hunk 4]
```diff
@@ -39,26 +43,36 @@
...
+                @input="onChange"
+                :type="color"
+              >Default</b-switch>
             </b-tooltip>
           </div>
         </a>
-        <component :is="connectorSocket" v-model="connectorDataChanged.socket"></component>
+        <component
+          v-model="connectorDataChanged.socket"
+          @input="onChange"
```
Comment (tjarbo) [inline packages/frontend/src/components/connectors/ConnectorPanel.vue:56]: Rename onChange to onInput

**Pull request merged**